client/events_test.go

```diff
@@ -146,23 +151,33 @@ func TestEvents(t *testing.T) {
 			}),
 		}
 
-		messages, errs := client.Events(context.Background(), eventsCase.options)
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second/2)
+		defer cancel()
 
-	loop:
-		for {
+		messages, errs := client.Events(ctx, eventsCase.options)
+
+		for ctx.Err() == nil {
 			select {
+			case <-ctx.Done():
+				if len(eventsCase.expectedEvents) != 0 {
+					t.Fatal("receive events timeout")
+				}
+
 			case err := <-errs:
 				if err != nil && err != io.EOF {
 					t.Fatal(err)
 				}
 
-				break loop
 			case e := <-messages:
-				_, ok := eventsCase.expectedEvents[e.ID]
-				if !ok {
-					t.Fatalf("event received not expected with action %s & id %s", e.Action, e.ID)
+				if e.ID != "" && !eventsCase.expectedEvents[e.ID] {
```

Comment:
Instead of checking for an empty ID here, consider checking the second value when receiving from a channel (Eg. `e, ok := < messages`) to see if we successfully read from the channel. This way we're not overloading what "empty message" means and are relying on channel semantics instead.